lib/oedipus_lex.rb

```diff
@@ -44,14 +44,7 @@ def to_ruby state, predicates, exclusive
         start_state == state or
         (state.nil? and predicates.include? start_state)
 
-      cond =
-        if exclusive or not start_state then
-          "when text = ss.scan(#{regexp}) then"
-        elsif start_state =~ /^:/ then
-          "when (state == #{start_state}) && (text = ss.scan(#{regexp})) then"
-        else
-          "when #{start_state} && (text = ss.scan(#{regexp})) then"
-        end
```

Comment:
I'd be happy to write more tests, but I think they are already covered?

Using the changes with RubyParser+Brakeman was a bigger test for me and I didn't see any issues.